x-pack/plugins/lens/public/editor_frame_service/error_helper.ts

```diff
@@ -26,54 +30,82 @@ const isRequestError = (e: Error | RequestError): e is RequestError => {
   return false;
 };
 
-interface ESError extends Error {
-  attributes?: { caused_by?: ElasticsearchErrorClause };
-}
-
-const isEsError = (e: Error | ESError): e is ESError => {
-  if ('attributes' in e) {
-    return e.attributes?.caused_by?.caused_by !== undefined;
-  }
-  return false;
-};
-
-function getNestedErrorClause({
+function getNestedErrorClauseWithContext({
   type,
   reason,
   caused_by: causedBy,
-}: ElasticsearchErrorClause): { type: string; reason: string } {
+  lang,
+  script,
+}: Reason): ReasonDescription[] {
+  if (!causedBy) {
+    return [{ type, reason }];
+  }
+  const [payload] = getNestedErrorClause(causedBy);
+  if (lang === 'painless') {
+    return [{ ...payload, context: { type: 'Painless script', reason: `"${script}"` || reason } }];
```

Comment:
should we i18n this? Seems like kind of a grey area...

Reply:
Thought about i18n here, but it would be the only message localized. The original string reports something like `"script error"` or similar, I'm just being a bit more precise.